Update VanillaInventoryCodeHooks.java

Draco18s opened the pull request:
Added check to prevent hopper from overfilling its own slots in the case of a subclass having a smaller max stack size.

Draco18s committed:
Update VanillaInventoryCodeHooks.java

Added check to prevent hopper from overfilling its own slots in the case of a subclass having a smaller max stack size.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=3294) <br/>All committers have signed the CLA.

Draco18s committed:
Update VanillaInventoryCodeHooks.java

Forgot the cast

liach commented:
~~That should be an issue on the modder's side, shouldn't it be?~~

Draco18s commented:
I have created a class, TileEntityWoodenHopper that extends TileEntityHopper.
When it extracts items from an inventory and puts them into its own slots (stack limit 16) it will "overfill" and delete the excess.  There is _no_ method I can override to fix this without a wholesale reimplementation of a dozen private hopper methods.
As an example, this works perfectly on 1.7.10:
https://github.com/Draco18s/HarderStuff/blob/master/src/main/java/com/draco18s/industry/entities/TileEntityWoodHopper.java
This does not work on 1.10.2:
https://github.com/Draco18s/ReasonableRealism/blob/master/src/main/java/com/draco18s/industry/entities/TileEntityWoodenHopper.java

Draco18s committed:
Update VanillaInventoryCodeHooks.java

Undo dumb. I don't need a cast. Working in a raw text editor without hinting messed me up making me think I'd forgotten something that was unnecessary.

liach commented:
~~Then you should make changes in the patch of vanilla hopper instead of here.~~

Draco18s commented:
I cannot, as the issue is here:
TileEntityHopper.java:398-401
`public static boolean captureDroppedItems(IHopper hopper) {`
`....Boolean ret = net.minecraftforge.items.VanillaInventoryCodeHooks.extractHook(hopper);`
`....if (ret != null) return ret;`
VanillaCodeHooks.extractHook runs, extracts an item, inserts it into the hopper (it gets deleted), method returns true, true is not null, captureDroppedItems returns true, hopper cooldown is set.  Done updating.

The vanilla code hooks are what is causing the incorrect behavior.

liach commented:
Actually this is good now.

Actuarius commented:
@mezz added labels [Bug]

Draco18s commented:
What is preventing this from getting merged?

Larsg310 commented:
You have a typo on `destStack.getMaxStatckSize()`.

Draco18s committed:
fixed typo

Draco18s commented:
Would've been nice to know that ten days ago.
Or someone could have merged and fixed it themselves.
Or edited the PR themselves.

jeffreykog commented:
Well, if you used a proper IDE to make this PR instead of the GitHub online editor you would have found that yourself

Draco18s commented:
The reason I haven't before now is because I've always been making simple changes.
Ironically I've spent most of the afternoon trying to do just that and it's actually not been an easy process.  "Clone and run setup" my ass.

LexManos commented:
Has this actually been tested that it compiles yet?
And yes the setup is that simple:
git clone
gradlew setupForge
eclipse -> Import existing projects.

Draco18s commented:
Thanks for that, Lex. Feel like updating the repo readme with that info? Because the current readme doesn't say that and everything I tried--following the existing instructions--didn't lead to a runnable project: either I got Forge (but no vanilla) or vanilla (but no Forge). That included double-clicking on the IntelliJ project file in the /projects directory.

Also, took me two tries even with your instructions, as the first time I didn't have "search for subprojects" checked and it broke the project definition. Three, if you count trying to import by Gradle Project, rather than Existing Projects (whoops).

Draco18s commented:
...How do I update the PR to point at the current revision now?  Because I didn't edit it through GitHub web, but made a push with GitHub Desktop, this PR doesn't point to the correct commit.

Daomephsta commented:
@Draco18s I assume you're referring to [this commit](https://github.com/Draco18s/MinecraftForge/commit/c3afa52db8f091db18c153aec39175b356dd393e). PR's will automatically update to the latest version of both branches, however that commit is on another branch.

Draco18s commented:
@Leviathan143 Ugh.  That's kind of what I thought, hence my confusion. That did not go where I wanted it to go. >:|  Trying to do things via touch screen remote desktop during the debate was less than ideal, but otherwise relatively productive.

Draco18s committed:
IDE Compile Check

Draco18s commented:
THERE. NOW IT WORKS.
I'M GOING TO GO GET DRUNK.
http://xkcd.com/1597/

Draco18s commented:
Ok, now what's preventing this from getting merged?

LexManos commented on src/main/java/net/minecraftforge/items/VanillaInventoryCodeHooks.java:
```diff
@@ -50,7 +50,7 @@ public static Boolean extractHook(IHopper dest)
                 {
                     ItemStack destStack = dest.getStackInSlot(j);
-                    if (destStack == null || destStack.stackSize < destStack.getMaxStackSize() && ItemHandlerHelper.canItemStacksStack(extractItem, destStack))
+                    if (destStack == null || destStack.stackSize < destStack.getMaxStackSize() && destStack.getMaxStackSize() < dest.getInventoryStackLimit() && ItemHandlerHelper.canItemStacksStack(extractItem, destStack))
```
Should be stackSize not getMaxStackSize.
Most items have a max stack of 64, and thats constant.
Stick it in a inventory that only allows... 5 and this logic would cause the item to never be able to be added at all.

Draco18s replied:
Thank you for noticing that, @LexManos I am not sure how I made that mistake.  Fixed now.

Draco18s committed:
Merge remote-tracking branch 'refs/remotes/MinecraftForge/1.10.x' into patch-1

Draco18s committed:
StackSize
